include/seqan3/utility/views/zip.hpp

```diff
@@ -6,26 +6,481 @@
+    zip::tuple_or_pair<std::ranges::sentinel_t<view_helper::maybe_const<Const, Views>>...> end_;
+
+    sentinel() = default;
+    constexpr sentinel(sentinel<!Const> i)
+        requires Const && (std::convertible_to<std::ranges::sentinel_t<Views>,
+                                               std::ranges::sentinel_t<view_helper::maybe_const<Const, Views>>> && ...)
+    : end_(std::move(i.end_))
+    {}
+
+    template <bool OtherConst>
+        requires (std::sentinel_for<std::ranges::sentinel_t<view_helper::maybe_const<Const, Views>>,
+                                    std::ranges::iterator_t<view_helper::maybe_const<OtherConst, Views>>> && ...)
+    friend constexpr bool operator==(iterator<OtherConst> const & x, sentinel const & y)
+    {
+        return [&] <size_t... N> (std::integer_sequence<size_t, N...>)
+        {
+            return ((std::get<N>(x.current_) == std::get<N>(y.end_)) || ...);
+        } (std::index_sequence_for<Views...>{});
+
+    }
+
+    template <bool OtherConst>
+        requires (std::sized_sentinel_for<std::ranges::sentinel_t<view_helper::maybe_const<Const, Views>>,
+                                          std::ranges::iterator_t<view_helper::maybe_const<OtherConst, Views>>> && ...)
+    friend constexpr std::common_type_t<std::ranges::range_difference_t<view_helper::maybe_const<OtherConst, Views>>...>
+        operator-(iterator<OtherConst> const & x, sentinel const & y)
+    {
+        return [&] <size_t... N> (std::integer_sequence<size_t, N...>)
+        {
+            return std::ranges::min({(zip::abs(std::get<N>(x.current_) - std::get<N>(y.end_)), ...)});
+        } (std::index_sequence_for<Views...>{});
+    }
+
+    template <bool OtherConst>
+        requires (std::sized_sentinel_for<std::ranges::sentinel_t<view_helper::maybe_const<Const, Views>>,
+                  std::ranges::iterator_t<view_helper::maybe_const<OtherConst, Views>>> && ...)
+    friend constexpr std::common_type_t<std::ranges::range_difference_t<view_helper::maybe_const<OtherConst, Views>>...>
+        operator-(sentinel const & y, iterator<OtherConst> const & x)
+    {
+        return - (x - y);
+    }
+};
+
+struct zip_fn
+{
+    template <typename urng_t>
+    constexpr auto operator()(urng_t && rng) const
+    {
+        return adaptor_from_functor{*this, std::forward<urng_t>(rng)};
+    }
+
+    template <typename... urng_ts>
+        requires (sizeof...(urng_ts) == 0)
+    constexpr auto operator()(urng_ts &&... ranges) const
+    {
+        return std::views::empty<seqan3::common_tuple<>>;
+    }
+
+    template <typename... urng_ts>
+        requires (sizeof...(urng_ts) > 1)
+    constexpr auto operator()(urng_ts &&... ranges) const
+    {
+        return zip_view{std::forward<urng_ts>(ranges)...};
+    }
+};
+
+} // namespace seqan3::detail
+//!\endcond
 
 namespace seqan3::views
 {
 
-/*!\brief A zip view
+/*!\brief A view adaptor that produces a tuple-like value of all passed views.
```

Comment:
```suggestion
/*!\brief A view adaptor that whose i-th element are tuple-like value of all i-ths elements of the passed views.
```
is `i-ths` a word?

Reply:
I find the suggestion more confusing :o

Reply:
taken from the std documentation :)

```suggestion
/*!\brief A view adaptor that takes several views and returns tuple-like values from every i-th element of each view.
```
?
I think your sentence sounds like it forms a tuple of views, not a tuple of view elements.

Reply:
taken from the std documentation :)

```suggestion
/*!\brief A view adaptor that takes several views and returns tuple-like values from every i-th element of each view.
```
?
I think your sentence sounds like it forms a tuple of views, not a tuple of view elements.